examples/providers/authenticator/src/main/java/org/keycloak/examples/authenticator/SecretQuestionAuthenticator.java

```diff
@@ -132,4 +138,9 @@ public void setRequiredActions(KeycloakSession session, RealmModel realm, UserMo
     public void close() {
 
     }
+
+    @Override
+    public SecretQuestionCredentialProvider getCredentialProvider(KeycloakSession session) {
+        return (SecretQuestionCredentialProvider)session.getProvider(CredentialProvider.class, "secret-question");
```

Comment:
Why not reuse the CREDENTIAL_TYPE variable, instead of the explicit string?
```suggestion
        return (SecretQuestionCredentialProvider)session.getProvider(CredentialProvider.class, CREDENTIAL_TYPE);
```

Reply:
It's actually not the same id -> "secret-question" vs "secret_question". But point taken, I'll create a static variable in the SecretQuestionCredentialProviderFactory, and use that.